### PR TITLE
7.0.0 - Support and Puppet 4.0

### PR DESCRIPTION
**Pull request by ffquintella**
This pull has the full code merged ... Sorry for the last problem.

### CODE REVIEW

**Comment by mkrakowitzer** [comment]
@ffquintella Thanks for this, I assume this is work in progress for now? Let me know if I can help with anything to get this PR completed.

File: manifests/config.pp [hunk 1]
```diff
@@ -61,16 +61,29 @@
...
+  if ($version[0] == '7' ){
+    file { "${jira::webappdir}/conf/server.xml":
+      content => template('jira/server7.xml.erb'),
+      mode    => '0600',
+      require => Class['jira::install'],
+      notify  => Class['jira::service'],
+    }
+  }else{
+    file { "${jira::webappdir}/conf/server.xml":
+      content => template('jira/server.xml.erb'),
```
Comment (igalic) [inline manifests/config.pp]: why don't we just make this template path a variable?

File: manifests/install.pp [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
   }
 
-  if ! defined(File[$jira::installdir]) {
+  if (!defined(File[$jira::installdir])) {
```
Comment (igalic) [inline manifests/install.pp]: unless?

File: manifests/install.pp [hunk 3]
```diff
@@ -40,10 +40,16 @@
     }
   }
 
-  $file = "atlassian-${jira::product}-${jira::version}.${jira::format}"
+  # https://www.atlassian.com/software/jira/downloads/binary/atlassian-jira-software-7.0.0-jira-7.0.0-x64.bin
+
+  if ($jira::version[0] == '7' ){
+    $file = "atlassian-jira-software-7.0.0-${jira::product}-${jira::version}${jira::format}"
+  }else {
+    $file = "atlassian-${jira::product}-${jira::version}${jira::format}"
```
Comment (igalic) [inline manifests/install.pp]: i think i'd prefer a selector here, but that's just me

File: manifests/config.pp (hunk 1 above)
Comment (mkrakowitzer) [inline manifests/config.pp]: I'd prefer we replaced the entire xml file template with augeas. That way we can keep up to date with all the latest modifications to the server.xml when we upgrade but still customize it.

File: manifests/install.pp (hunk 2 above)
Comment (ffquintella) [inline manifests/install.pp]: Well this is taste. ;-) I like it better this way.

File: manifests/config.pp (hunk 1 above)
Comment (ffquintella) [inline manifests/config.pp]: It's possible to do it. Or maybe just instrumentate the template to be just one. But this way it's  working for me. I can adjust it on another push latter.

File: manifests/install.pp (hunk 3 above)
Comment (ffquintella) [inline manifests/install.pp]: I usualy use selectors when there are more then 2 options ... but thinking on the future yes this could be usefull.

**Comment by whitewolf712** [comment]
Hi! I'm try to install Jira 7 with my Puppet 3 (3.8.1 on master and 3.7.2 on agent) and get an error: 
```
jira::version is not a hash or array when accessing it with 0 at /etc/puppet/environments/jira7/modules/jira/manifests/install.pp:45 on node jira7-app.
```

How do you feel about use versioncmp function instead first character comparing? On files config.pp and install.pp

**Comment by igalic** [comment]
@whitewolf712 are you running with future parser enabled?
@ffquintella i think we should regex-check whether the string begins with '7':
```puppet
$file = $jira::version? {
  /^7/    => "atlassian-jira-software-7.0.0-${jira::product}-${jira::version}${jira::format}",
  default => "atlassian-${jira::product}-${jira::version}${jira::format}"
```

**Comment by whitewolf712** [comment]
@igalic, probaly future parser is disabled, i don't know about this feature before read your comment. I try use this later. With regexp code looks readable =)

What about describe future parser requirement on module documentation?

**Comment by ffquintella** [comment]
Yes it's  enabled... I'm tring to get the code fully compliant with the new features.

File: manifests/install.pp (hunk 3 above)
Comment (jared-di) [inline manifests/install.pp]: There is more than 2 options..

JIRA Core:
atlassian-jira-core-7.0.0.tar.gz
JIRA Software (JIRA + Agile):
atlassian-jira-software-7.0.0-jira-7.0.0.tar.gz
JIRA Service Desk (JIRA + Service Desk):
atlassian-servicedesk-3.0.0-jira-7.0.0.tar.gz

**Comment by Aricg** [comment]
Any progress on this? I am tasked with getting this module working with jira7, looks like If this were merged I would already be done.

**Comment by igalic** [comment]
versioncmp()!

**Comment by ffquintella** [comment]
There is a small bugfix on download I updated today

**Comment by igalic** [comment]
@ffquintella nice seeing progress here!
please note that travis complains about:
```
manifests/config.pp:64:variable_scope:WARNING:top-scope variable being used without an explicit namespace
manifests/config.pp:77:trailing_whitespace:ERROR:trailing whitespace found
```

and will probably complain even more once you fix those :b
this will need a propper [rebase](https://www.git-scm.com/book/en/v2/Git-Branching-Rebasing).
Please note that a rebase will be easier if you use the [github flow](https://guides.github.com/introduction/flow/index.html) work-flow. A nice-step-by-step how to can be found here: https://github.com/puppet-community/puppet-jira/pull/99#issuecomment-152528108

**Comment by Aricg** [comment]
Hi @ffquintella looks like this one should be abandoned in favor of you other pull request https://github.com/puppet-community/puppet-jira/pull/106

**Comment by jyaworski** [comment]
Closing in favor of #106

**Pull request closed without merging**